packages/merge-styles/src/shadowConfig.ts

```diff
@@ -24,10 +25,14 @@ export const makeShadowConfig = (stylesheetKey: string, inShadow: boolean, windo
   };
 };
 
-export const isShadowConfig = (obj: unknown): obj is ShadowConfig => {
-  if (!obj) {
+export const isShadowConfig = (value: unknown): value is ShadowConfig => {
+  if (!(value && isRecord(value))) {
     return false;
   }
 
-  return (obj as ShadowConfig).__isShadowConfig__ === true;
```

Comment:
this was a weak check enough check from Runtime POV, now it should be more robust

Reply:
Agree that it is not the strongest test but this function is also called frequently so the idea was to get it a simple as possible to avoid runtime perf hits. All that said, the change doesn't seem to have moved the perf numbers so having a more robust solution works for me :)